packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -36,23 +37,26 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
       !options.disableMultipleSelection && isMultipleKeyPressed;
   }, [isMultipleKeyPressed, options.disableMultipleSelection]);
 
-  const getSelectedRows = React.useCallback((): GridRowModel[] => {
-    // TODO replace with selector
-    return Object.keys(gridState.selection).map((id) => apiRef.current.getRowFromId(id));
-  }, [apiRef, gridState.selection]);
+  const getSelectedRows = React.useCallback(
+    () => selectedGridRowsSelector(apiRef.current.getState()),
+    [apiRef],
+  );
 
   const selectRowModel = React.useCallback(
-    (row: GridRowModel, allowMultipleOverride?: boolean, isSelected?: boolean) => {
+    (id: GridRowId, row: GridRowModel, allowMultipleOverride?: boolean, isSelected?: boolean) => {
+      const idAsNumber = Number(id);
+      const finalId = Number.isNaN(idAsNumber) ? id : idAsNumber;
```

Comment:
Selecting a row with the keyboard will always call this method with a string, because `getIdFromRowElem` relays in the `data-id` attribute. I might move this cast to `getIdFromRowElem`.

https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts#L60-L68

Reply:
You need to refactor this function to use the params id of `handleCellKeyDown`. So you already have the id and don't need to retrieve it from the element